controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -59,10 +65,13 @@ func (s *ClusterUrlMonitorSupplement) EnsureServiceMonitorExists() error {
 	if err != nil {
 		return err
 	}
+	if s.ClusterID == "" {
+		s.ClusterID = templates.GetClusterID(s.Client)
+	}
```

Comment:
won't `s.ClusterID` be empty on every reconcile?

Reply:
Only after the first reconcile and in case an error happend during the GetClusterID. In this case it will try to fetch the ClusterID again

Reply:
And Supplement is just allocated once [here](https://github.com/openshift/route-monitor-operator/blob/4abfff4b9a4ba5552c48d29db47e965c2e050b4c/main.go#L102), so values are maintained between reconcile loops

Reply:
ok no you are right, its not for the clusterurlmonitor_supplement but just for the routemonitor.
A new clusterurl Supplement is allocated on each Reconcile. Thats a bit unfortunate. However, I could move the ClusterID fetching and it would have the same effect. However, what do you think. Is this approach ok, or should we go with altering the BlackBoxExporter Deployment CR ?

Reply:
I don't mind too much at which object it is.